tinylog-impl/src/test/java/org/tinylog/writers/SyslogWriterTest.java

```diff
@@ -0,0 +1,311 @@
+	 * 
+	 * @throws Exception Failed.
+	 */
+	@Test
+	public void sendTcpSyslogMessageWithIdentification() throws Exception {
+		TcpSyslogServer server = new TcpSyslogServer(TEST_PORT_NUMBER);
+		server.start();
+		SyslogFacility facility = SyslogFacility.KERN;
+		SyslogSeverity severity = SyslogSeverity.DEBUG;
+		String identification = "SyslogWriterTest";
+		Map<String, String> properties = new HashMap<>();
+		properties.put("format", "{message}");
+		properties.put("protocol", "tcp");
+		properties.put("port", TEST_PORT_NUMBER.toString());
+		properties.put("facility", facility.toString());
+		properties.put("severity", severity.toString());
+		properties.put("identification", identification);
+		SyslogWriter writer = new SyslogWriter(properties);
+		writer.write(LogEntryBuilder.empty().message(TEST_MESSAGE).create());
+		Thread.sleep(250);
+		assertThat(server.getLastMessage()).isEqualTo(getExpectedSyslogMessage(TEST_MESSAGE, facility, severity, identification));
+		server.shutdown();
+	}
+
+	/**
+	 * Sends tcp message with different levels and verifies it is received.
+	 * 
+	 * @throws Exception Failed.
+	 */
+	@Test
+	public void sendLevelControlledTcpSyslogMessage() throws Exception {
+		TcpSyslogServer server = new TcpSyslogServer(TEST_PORT_NUMBER);
+		server.start();
+		SyslogFacility facility = SyslogFacility.AUTH;
+		Map<String, String> properties = new HashMap<>();
+		properties.put("format", "{message}");
+		properties.put("protocol", "tcp");
+		properties.put("port", TEST_PORT_NUMBER.toString());
+		properties.put("facility", facility.toString());
+		SyslogWriter writer = new SyslogWriter(properties);
+		LogEntryBuilder log = LogEntryBuilder.empty().message(TEST_MESSAGE);
+		log.level(Level.DEBUG);
+		writer.write(log.create());
+		Thread.sleep(250);
+		assertThat(server.getLastMessage()).isEqualTo(getExpectedSyslogMessage(TEST_MESSAGE, facility, SyslogSeverity.DEBUG, ""));
+		log.level(Level.ERROR);
+		writer.write(log.create());
+		Thread.sleep(250);
+		assertThat(server.getLastMessage()).isEqualTo(getExpectedSyslogMessage(TEST_MESSAGE, facility, SyslogSeverity.ERROR, ""));
+		log.level(Level.INFO);
+		writer.write(log.create());
+		Thread.sleep(250);
+		assertThat(server.getLastMessage()).isEqualTo(getExpectedSyslogMessage(TEST_MESSAGE, facility, SyslogSeverity.INFO, ""));
+		log.level(Level.OFF);
+		writer.write(log.create());
+		Thread.sleep(250);
+		assertThat(server.getLastMessage()).isEqualTo(getExpectedSyslogMessage(TEST_MESSAGE, facility, SyslogSeverity.EMERG, ""));
+		log.level(Level.TRACE);
+		writer.write(log.create());
+		Thread.sleep(250);
+		assertThat(server.getLastMessage()).isEqualTo(getExpectedSyslogMessage(TEST_MESSAGE, facility, SyslogSeverity.DEBUG, ""));
+		log.level(Level.WARN);
+		writer.write(log.create());
+		Thread.sleep(250);
+		assertThat(server.getLastMessage()).isEqualTo(getExpectedSyslogMessage(TEST_MESSAGE, facility, SyslogSeverity.WARNING, ""));
+		server.shutdown();
+	}
+}
```

Comment:
Please add an empty line at the end of the class body.

Reply:
Updated